File: app/src/main/java/com/BlizzardArmory/util/MetricConversion.java
package com.BlizzardArmory.util;

import android.app.Activity;

/**
 * The type Metric conversion.
 */
public class MetricConversion {

    /**
     * Gets dp metric.
     *
     * @param size     the size
     * @param activity the activity
     * @return the dp metric
     */
    public static int getDPMetric(int size, Activity activity) {
        final float scale = activity.getResources().getDisplayMetrics().density;
        return (int) (size * scale + 0.5f);
    }
}

File: app/src/main/java/com/BlizzardArmory/util/MetricConversion.kt
package com.BlizzardArmory.util

import android.content.Context

/**
 * The type Metric conversion.
 */
object MetricConversion {
    /**
     * Gets dp metric.
     *
     * @param size     the size
     * @param context the activity
     * @return the dp metric
     */
    @JvmStatic
    fun getDPMetric(size: Int, context: Context): Int {
        val scale = context.resources.displayMetrics.density
        return (size * scale + 0.5f).toInt()
    }
}